server/irmaserver/sessions.go

```diff
@@ -16,46 +21,50 @@ import (
 )
 
 type session struct {
-	sync.Mutex
-	locked bool
+	//TODO: check if we can get rid of this Mutex for Redis
+	sync.Mutex `json:-`
+	//TODO: note somewhere that state with redis will not support sse for the moment
+	sse         *sse.Server
+	locked      bool
+	sessions    sessionStore
+	conf        *server.Configuration
+	request     irma.SessionRequest
+	context     context.Context
+	toBeUpdated bool
 
-	action             irma.Action
-	token              string
-	clientToken        string
-	version            *irma.ProtocolVersion
-	rrequest           irma.RequestorRequest
-	request            irma.SessionRequest
-	legacyCompatible   bool // if the request is convertible to pre-condiscon format
-	implicitDisclosure irma.AttributeConDisCon
-
-	status        server.Status
-	prevStatus    server.Status
-	sse           *sse.Server
-	responseCache responseCache
-
-	lastActive time.Time
-	result     *server.SessionResult
-
-	kssProofs map[irma.SchemeManagerIdentifier]*gabi.ProofP
+	sessionData
+}
 
-	conf     *server.Configuration
-	sessions sessionStore
+type sessionData struct {
+	Action             irma.Action
+	Token              string
+	ClientToken        string
+	Version            *irma.ProtocolVersion `json:",omitempty"`
+	Rrequest           irma.RequestorRequest
+	LegacyCompatible   bool // if the request is convertible to pre-condiscon format
+	ImplicitDisclosure irma.AttributeConDisCon
+	Status             server.Status
+	PrevStatus         server.Status
+	ResponseCache      responseCache
+	LastActive         time.Time
+	Result             *server.SessionResult
+	KssProofs          map[irma.SchemeManagerIdentifier]*gabi.ProofP
```

Comment:
Sounds like a good alternative! I'm considering solving the Redis implementation with HSETs instead of the JSON in general. I will come back to this comment when I have worked out how to use Redis best in general.